crates/api/src/transport.rs

```diff
@@ -245,59 +250,133 @@ impl TxImpHnd {
-                    tracing::debug!(?space_id, ?peer_url, ?message_type, ?module_id, "All agents at peer are blocked, message will be dropped.");
-                    self.incr_blocked_message_count(peer_url.clone(), space_id);
-                    return Ok(false);
-                }
-                Ok(true)
-            }
-            None => {
-                tracing::error!(
+        let is_blocked = is_peer_blocked(
+            self.space_map.clone(),
+            self.blocked_message_counts.clone(),
+            peer_url,
+            &space_id,
+            module_id,
+            false,
+        )?;
+        Ok(!is_blocked)
+    }
+}
+
+impl std::fmt::Debug for TxImpHnd {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f,"TxImpHnd {{ handler: {:?}, space_map: [{} entries], mod_map: [{} entries] }}", self.handler, self.space_map.lock().unwrap().len(), self.mod_map.lock().unwrap().len() )
+    }
+}
+
+/// Check whether all agents associated with the given peer url are blocked
+/// for a space and increase the message blocks count by one if they are.
+fn is_peer_blocked(
+    space_map: SpaceMap,
+    message_blocks_map: MessageBlocksMap,
+    peer_url: &Url,
+    space_id: &SpaceId,
+    module_id: &Option<String>,
+    outgoing: bool,
+) -> K2Result<bool> {
+    match space_map.lock().expect("poisoned").get(space_id) {
+        Some(space_handler) => {
+            let space_handler: Arc<dyn TxSpaceHandler> = space_handler.clone();
+            let all_blocked = space_handler.are_all_agents_at_url_blocked(peer_url).inspect_err(|e| tracing::warn!(?space_id, ?peer_url, ?module_id, "Failed to check whether all agents are blocked, peer connection will be closed: {e}"))?;
+            if all_blocked {
+                tracing::debug!(
                     ?space_id,
                     ?peer_url,
                     ?module_id,
-                    ?message_type,
-                    "No space handler found. Message will be dropped."
+                    "All agents at peer are blocked, message will be dropped."
                 );
-                Ok(false)
+                if outgoing {
+                    incr_blocked_message_count_outgoing(
+                        message_blocks_map,
+                        peer_url.clone(),
+                        space_id,
+                    );
+                } else {
+                    incr_blocked_message_count_incoming(
+                        message_blocks_map,
+                        peer_url.clone(),
+                        space_id,
+                    );
+                }
+                return Ok(true);
             }
+            Ok(false)
+        }
+        None => {
+            tracing::error!(
+                ?space_id,
+                ?peer_url,
+                ?module_id,
+                "No space handler found. Message will be dropped."
+            );
+            Ok(true)
         }
-    }
-
```

Comment:
This method did not distinguish between incoming and outgoing message blocks so it gets replaced.